### PR TITLE
refactor(connector-fabric): deployContractGoSourceV1 uses Fabric v2.5.6

### PR DESCRIPTION
**Pull request by petermetz**
The deployContractGoSourceV1() method now assumes that the underlying
test ledger is Fabric 2.5 (current LTS).

This will allow us to upgrade the contracts that are being used by the
Supply chain app to Fabric 2.x from Fabric 1.x which will then implicitly
fix a large number of other issues at the same time.

This change is part of laying the foundation for that follow-up work.

Primary changes:
-----------------

1. Added a new, standalone utility function to deploy go source contracts
with the name of `deployContractGoSourceImplFabricV256()`.
2. The code of this function was derived from the original Fabric v1
compatible deployContractGoSourceV1 method of the Fabric connector.
3. 2 organizations are supported for deployment via the endpoint.
4. The endpoint is only used by the supply chain app example at the moment
and there is no test coverage of it due to dependencies that will be
resolved in a follow-up pull request that is coming soon.

Secondary changes:
1. Also extracted the SSH execution function from the fabric connector
into a standalone function that can be used without having to have a
Fabric connector instance created first.
2. Also extracted/abstracted some logic into a utility function for
similar reasons that is used to replace logging configuration environment
variables in shell commands that we use to perform contract deployment
onto the Fabric test ledgers.

Depends on #3054

Signed-off-by: Peter Somogyvari <peter.somogyvari@accenture.com>

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

File: packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/deploy-contract-go-source/deploy-contract-go-source-impl-fabric-v2-5-6.ts [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    const inferredModuleName = path.basename(req.goSource.filename, ".go");
+    log.debug(`Inferred module name: ${inferredModuleName}`);
+    const ccName = req.moduleName || inferredModuleName;
+    log.debug(`Determined ChainCode name: ${ccName}`);
+
+    const remoteDirPath = path.join(cliContainerGoPath, "src/", ccName);
+    log.debug(`Remote dir path on CLI container: ${remoteDirPath}`);
+
+    const localFilePath = path.join(tmpDirPath, req.goSource.filename);
+    await fs.writeFile(localFilePath, req.goSource.body, "base64");
```
[the remark this answers is not in the document]
Comment (petermetz) [reply packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/deploy-contract-go-source/deploy-contract-go-source-impl-fabric-v2-5-6.ts:77]: The contract deployment endpoint is for development use **only** and therefore it is considered acceptable to have such a vulnerability in place for it given that is meant to be used for/by automated test cases.

With that said, it should be moved out of the production codebase (from the connector) eventually.

**Review by outSH — APPROVED** [review]

**Review by outSH — APPROVED** [review]
LGTM, squash the commits before the merge (I've disabled an auto-merge)

**Comment by petermetz** [comment]
> LGTM, squash the commits before the merge (I've disabled an auto-merge)

@outSH Oops, sorry and thank you very much for being on the lookout! I should be more careful with the auto-merge feature for pull requests that have parents!

**Pull request merged**